Core/clim-core/panes/layout-protocol.lisp

```diff
@@ -473,13 +469,30 @@ which changed during the current execution of CHANGING-SPACE-REQUIREMENTS.
 
 (defmethod change-space-requirements ((pane layout-protocol-mixin)
                                       &key resize-frame &allow-other-keys)
-  (when-let ((parent (sheet-parent pane)))
-    (change-space-requirements parent :resize-frame resize-frame)))
+  ;; do nothing here
+  nil)
 
 (defmethod change-space-requirements :after ((pane layout-protocol-mixin)
                                              &key resize-frame &allow-other-keys)
-  (declare (ignore resize-frame))
-  (note-space-requirements-changed (sheet-parent pane) pane))
+  (when-let ((parent (sheet-parent pane)))
+    (if resize-frame
+        ;; From Spec 29.3.4: "If resize-frame is true, then
+        ;; layout-frame will be invoked on the frame". Here instead of
+        ;; call directly LAYOUT-FRAME, we call
+        ;; CHANGE-SPACE-REQUIREMENTS on the parent and it travels all
+        ;; the way up to the top-level-sheet-pane which then invokes
+        ;; the layout protocol calling LAYOUT-FRAME. The rationale of
+        ;; this is:
+        ;; 1. we can't call (LAYOUT-FRAME (PANE-FRAME pane)) on a
+        ;;   menu because with the actual implementation of menu it
+        ;;   will layout the main application and not the menu frame.
+        ;; 2. we automatically clear the cached values of
+        ;;    space-requirements for the involved panes.
+        ;; -- admich 2020-08-11
+        (if (top-level-sheet-pane-p pane)
+            (note-space-requirements-changed (sheet-parent pane) pane)
```

Comment:
top-levle-sheet's parent is the graft, is that right that we call the function on the graft? Isn't this supposed to be
(layout-frame frame)?
or even
```
(if (top-level-sheet-pane-p pane)
   (if resize-frame (layout-frame pane) (layout-frame pane width height))
   (note-space-requirements-changed parent pane))
```

I don't think that conditions here are right.

Reply:
Note that this branch is executed when :resize-frame is t. It would be correct to simply call (layout-frame (pane-frame frame) here for all the pane, but in McCLIM for the menu frame (pane-frame frame) return the main application and not the menu. I explain this in the comment.

When change-space-requirements on a top-level-sheet it is correct to call note-space-requirements-changed because then the parent (the graft) ask space to the window manager.

The layout-frame function is called in the primary method of change-space-requirements on top-level-sheet.